src/server_manager/web_app/location_formatting.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {CloudLocation, GeoId, DataCenterMap, countryCode, DataCenterId} from '../model/location';
+
+/**
+ * Returns the localized place name, or the data center ID if the location is
+ * unknown.
+ */
+export function getShortName(location: CloudLocation,
+    localize: (id: string) => string): string {
+  if (!location?.geoId) {
+    return location?.id ?? '';
+  }
+  return localize(`geo-${location.geoId.toLowerCase()}`);
+}
+
+/**
+ * Returns the localized country name, or "" if the country is unknown or
+ * unnecessary.
+ */
+export function localizeCountry(geoId: GeoId, language: string): string {
+  if (!geoId) {
+    return '';
+  }
+  const cc = countryCode(geoId);
+  if (cc === geoId) {
```

Comment:
This is application logic. Move this to the caller. Keep this method as simple country code, lang -> name

Reply:
Done.  I added `countryIsRedundant` method to `GeoLocation` to encapsulate this equality check.